src/System.ServiceProcess.ServiceController/tests/System.ServiceProcess.ServiceController.TestService/TestService.cs

```diff
@@ -29,66 +33,99 @@ public TestService(string serviceName)
             // We cannot easily test these so disable the events
             this.CanHandleSessionChangeEvent = false;
             this.CanHandlePowerEvent = false;
-        }
 
-        public static string GetLogPath(string serviceName)
-        {
-            return typeof(TestService).Assembly.Location + "." + serviceName + ".log";
+            this.Server = new NamedPipeServerStream(serviceName);
+            serverStarted = this.Server.WaitForConnectionAsync();
         }
 
+        private NamedPipeServerStream Server { get; }
+
         protected override void OnContinue()
         {
-            WriteLog(nameof(OnContinue));
             base.OnContinue();
+            var task = Task.Run(() => WriteStreamAsync(PipeMessageByteCode.Continue));
```

Comment:
>Task.Run(() => WriteStreamAsync(PipeMessageByteCode.Continue)); [](start = 23, length = 63)

You don't need Task.Run here the method itself returns a Task.

Reply:
Similar to other locations

---
In reply to: [164004333](https://github.com/dotnet/corefx/pull/26524#discussion_r164004333) [](ancestors = 164004333)